src/components/CharacterAssignment.jsx

```diff
@@ -0,0 +1,229 @@
+import React, { useEffect, useState } from 'react';
+import { DragDropContext } from 'react-beautiful-dnd';
+import ActorColumn from './ActorColumn.jsx'
+import UnassignedColumn from './UnassignedColumn';
+
+// Entire Drag and Drop context for assigning characters to actors
+const CharacterAssignment = ({actors, currentScript}) => {
+  const [columnOrder, setColumnOrder] = useState([])
+  const [columns, setColumns] = useState({
+    'unassignedCharacters': {
+      id:'unassignedCharacters',
+        title: 'Unassigned Characters',
+        characterIds: [],
+        lineCount: 0
+    }});
+  const [characters, setCharacters] = useState()
+
+  // Create the actor columns initially and edit columns when actors are added or removed
+  useEffect(() => {createOrUpdateActorColumns(actors)}, [actors]);
+
+  // get the character data on the initial load and when the script option changes
+  useEffect(() => { (async () => { // useEffect cannot take an async function. Must wrap async in regular function
+    if (currentScript === null) {
+      return;
+    }
+    try {
+      const response = await fetch('/api/script/' + currentScript + '/characters');
+
+      if (response.ok) {
+        const fetchedCharacters = await response.json()
+        resetActors()
```

Comment:
If we set state multiple times, does that mean we trigger multiple renders in quick succession, or do we just render once _after_ all the useEffect code is finished running?